### DISCUSSION TITLE
[Feature] `yarn npm publish` get token from environment variable

### DISCUSSION THREAD

**Discussion by nodece** [yarnpkg/berry#2985] [Q&A] [answered]
**Describe the user story**

I want to use `yarn npm publish` to publish my package in CI, I'm not sure this command whether support get token from environment variable.

**Describe the solution you'd like**

Support get token from environment variable.

**Comment by merceyz** [comment, marked as answer]
Yarn supports loading settings from the environment, it's one of the first things documented here https://yarnpkg.com/configuration/yarnrc, for this specific case it would be
```
YARN_NPM_AUTH_TOKEN="<token>" yarn npm publish
```

https://yarnpkg.com/configuration/yarnrc#npmAuthToken
